diff -up mysql-5.6.24/storage/perfschema/pfs_account.cc.p57 mysql-5.6.24/storage/perfschema/pfs_account.cc
--- mysql-5.6.24/storage/perfschema/pfs_account.cc.p57	2015-03-25 17:34:52.000000000 +0100
+++ mysql-5.6.24/storage/perfschema/pfs_account.cc	2015-04-09 06:53:12.383168058 +0200
@@ -201,6 +201,13 @@ static void set_account_key(PFS_account_
   key->m_key_length= ptr - &key->m_hash_key[0];
 }
 
+#if defined(__aarch64__)
+PFS_account *
+find_or_create_account(PFS_thread *thread,
+		       const char *username, uint username_length,
+		       const char *hostname, uint hostname_length) __attribute__((optimize (1)));
+#endif
+
 PFS_account *
 find_or_create_account(PFS_thread *thread,
                          const char *username, uint username_length,
diff -up mysql-5.6.24/storage/perfschema/pfs_digest.cc.p57 mysql-5.6.24/storage/perfschema/pfs_digest.cc
--- mysql-5.6.24/storage/perfschema/pfs_digest.cc.p57	2015-04-09 06:53:12.386168058 +0200
+++ mysql-5.6.24/storage/perfschema/pfs_digest.cc	2015-04-09 06:56:20.385213275 +0200
@@ -170,6 +170,14 @@ static LF_PINS* get_digest_hash_pins(PFS
   return thread->m_digest_hash_pins;
 }
 
+#if defined(__aarch64__)
+PFS_statement_stat*
+find_or_create_digest(PFS_thread *thread,
+                      const sql_digest_storage *digest_storage,
+                      const char *schema_name,
+                      uint schema_name_length) __attribute__((optimize (1)));
+#endif
+
 PFS_statement_stat*
 find_or_create_digest(PFS_thread *thread,
                       const sql_digest_storage *digest_storage,
diff -up mysql-5.6.24/storage/perfschema/pfs_host.cc.p57 mysql-5.6.24/storage/perfschema/pfs_host.cc
--- mysql-5.6.24/storage/perfschema/pfs_host.cc.p57	2015-03-25 17:34:52.000000000 +0100
+++ mysql-5.6.24/storage/perfschema/pfs_host.cc	2015-04-09 06:53:12.386168058 +0200
@@ -193,6 +193,11 @@ static void set_host_key(PFS_host_key *k
   key->m_key_length= ptr - &key->m_hash_key[0];
 }
 
+#if defined(__aarch64__)
+PFS_host *find_or_create_host(PFS_thread *thread,
+                              const char *hostname, uint hostname_length) __attribute__((optimize (1)));
+#endif
+
 PFS_host *find_or_create_host(PFS_thread *thread,
                               const char *hostname, uint hostname_length)
 {
diff -up mysql-5.6.24/storage/perfschema/pfs_instr.cc.p57 mysql-5.6.24/storage/perfschema/pfs_instr.cc
--- mysql-5.6.24/storage/perfschema/pfs_instr.cc.p57	2015-03-25 17:34:52.000000000 +0100
+++ mysql-5.6.24/storage/perfschema/pfs_instr.cc	2015-04-09 06:53:12.388168058 +0200
@@ -1199,6 +1199,11 @@ LF_PINS* get_filename_hash_pins(PFS_thre
   @param create                       create a file instance if none found
   @return a file instance, or NULL
 */
+#if defined(__aarch64__)
+PFS_file*
+find_or_create_file(PFS_thread *thread, PFS_file_class *klass,
+                    const char *filename, uint len, bool create) __attribute__((optimize (1)));
+#endif
 PFS_file*
 find_or_create_file(PFS_thread *thread, PFS_file_class *klass,
                     const char *filename, uint len, bool create)
diff -up mysql-5.6.24/storage/perfschema/pfs_instr_class.cc.p57 mysql-5.6.24/storage/perfschema/pfs_instr_class.cc
--- mysql-5.6.24/storage/perfschema/pfs_instr_class.cc.p57	2015-03-25 17:34:52.000000000 +0100
+++ mysql-5.6.24/storage/perfschema/pfs_instr_class.cc	2015-04-09 06:53:12.389168058 +0200
@@ -1233,6 +1233,11 @@ static int compare_keys(PFS_table_share
   @param share                        table share
   @return a table share, or NULL
 */
+#if defined(__aarch64__)
+PFS_table_share* find_or_create_table_share(PFS_thread *thread,
+                                            bool temporary,
+                                            const TABLE_SHARE *share) __attribute__((optimize (1)));
+#endif
 PFS_table_share* find_or_create_table_share(PFS_thread *thread,
                                             bool temporary,
                                             const TABLE_SHARE *share)
diff -up mysql-5.6.24/storage/perfschema/pfs_user.cc.p57 mysql-5.6.24/storage/perfschema/pfs_user.cc
--- mysql-5.6.24/storage/perfschema/pfs_user.cc.p57	2015-03-25 17:34:52.000000000 +0100
+++ mysql-5.6.24/storage/perfschema/pfs_user.cc	2015-04-09 06:53:12.389168058 +0200
@@ -193,6 +193,12 @@ static void set_user_key(PFS_user_key *k
   key->m_key_length= ptr - &key->m_hash_key[0];
 }
 
+#if defined(__aarch64__)
+PFS_user *
+find_or_create_user(PFS_thread *thread,
+                    const char *username, uint username_length)  __attribute__((optimize (1)));
+#endif
+
 PFS_user *
 find_or_create_user(PFS_thread *thread,
                     const char *username, uint username_length)
