diff -Naurp a/storage/perfschema/unittest/CMakeLists.txt b/storage/perfschema/unittest/CMakeLists.txt
--- a/storage/perfschema/unittest/CMakeLists.txt	2016-05-25 00:50:36.000000000 -0500
+++ b/storage/perfschema/unittest/CMakeLists.txt	2016-06-03 17:49:20.830635665 -0500
@@ -37,7 +37,6 @@ SET(tests
  pfs_instr_class
  pfs_instr_class-oom
  pfs_instr
- pfs_instr-oom
  pfs_account-oom
  pfs_host-oom
  pfs_user-oom
